### ISSUE TITLE
Razer Chroma RGB API Support

### ISSUE THREAD

**Issue by eVenent** [Paliverse/DualSenseX#83]
**Is your feature request related to a problem? Please describe.**
DualSense is enabling PS5 exclusive features for PC. One feature is still neglected - lightbar. It can have much bigger potential. It is difficult to force game developers to implement lightbar support for Playstation controllers. But many of them are implementing [Razer Chroma RGB support](https://www.razer.com/chroma-workshop#--games) for keyboards and mice and different PC peripherials what can help with enabling advanced support for Dualsense lightbar.

**Describe the solution you'd like**
I would like request feature that DualSenseX lightbar control would be compatible with [Razer Chroma RGB API](https://developer.razer.com/works-with-chroma/download/). In this way users would receive decent lightbar support without hacking games or asking developers to implement special native support for Dualsense. Configs are already implemented in supported games. Dualsense have to just read signals to change LED color from Razer Chroma signals.

**Describe alternatives you've considered**
There is no alternative for Dualsense's lightbar support for Razer Chroma RGB yet. Just few games with [native support.](https://www.pcgamingwiki.com/wiki/Controller:DualShock_4#Games_with_Light_Bar_support)
Although [Project Aurora](https://github.com/antonpup/Aurora) enables it for Dualshock 4 (but it's difficult and not always working). We need application like DualSenseX which has better interface and is easy to use.

**Additional context**
I would like to have visual feedback on actions which I am doing in games. [MONSTER HUNTER RISE](https://www.razer.com/chroma-workshop#--games--look--Y2FwY29tJTIwY28uJTJDJTIwbHRkLi9tb25zdGVyJTIwaHVudGVyJTIwcmlzZQ_-__-_) is perfect example how lightbar feedback can looks like. LED is:
* Red breathing effects when health is low
* Green blinking effects when recovering health
* Blue blinking effects when monsters’ parts are broken
* Rainbow effects when starting new quest
* Blue blinking effects when other player joins your quest with Join Requests
* Various blinking effects when player is under various blight status

**Comment by Paliverse** [comment]
Does implementing this api have to be a game by game basis when it comes to changing the Touchpad LED?

**Comment by eVenent** [comment]
No, no. It have to be a "supported device connector". As game DualSenseX will send signals to change color of different devices, but we need to be receiver of these signals, not sender.

I am using it in Yeelight led strip with Razer Chroma application (so it's 3rd party device), which works nearly the same as Dualsense in this case - only one color possible at the time. I had to install Yeelight application which is bridge between Razer Chroma application. Here is video how looks whole process of [pairing](https://www.yeelight.com/pages/razer.html#eighth).
1. I had to run application which is device connector.
2. When application is compatible with Razer Chroma API it appears in "supported devices" menu and I had to check, that I want to use it with Razer Chroma.
3. Now if game is compatible with Razer Chroma, game is talking to application (Razer Synapse) and application is sending signals to enabled supported devices. Game is changing colors of device.

Signal is going like this:
Game with Chroma support -> Razer Synapse -> Device connector app (DualSenesX) -> Device changing color (DualSense)
With Yeelight led strip I did not observe any lags, so there is no issue with many connectors. All works fast and as expected. 

I believe it would be next big thing for DualSense support on PC.

**Comment by Paliverse** [comment]
Okay I understand what you mean now. Only issue I see so far is the the API is written in C++ Language while DSX is written with C#. I’ll dig around and see what I can find to implement the API. Should be doable to get this feature in

**Comment by eVenent** [comment]
Thank you! The feature which is worth to wait for.
